docs/template_plugin/src/template_infer_request.hpp

```diff
@@ -57,17 +61,18 @@ class TemplateInferRequest : public InferenceEngine::InferRequestInternal {
         numOfStages
     };
 
-    std::array<InferenceEngine::ProfilingTask, numOfStages> _profilingTask;
+    std::array<InferenceEngine::ProfilingTask, numOfStages>             _profilingTask;
+    // for performance counters
+    std::array<std::chrono::duration<float, std::micro>, numOfStages>   _durations;
 
-    InferenceEngine::BlobMap                                _inputsNCHW;
-    InferenceEngine::BlobMap                                _outputsNCHW;
+    InferenceEngine::BlobMap                                _networkInputBlobs;
+    InferenceEngine::BlobMap                                _networkOutputBlobs;
+    ngraph::ParameterVector                                 _parameters;
+    ngraph::ResultVector                                    _results;
 
-    // for performance counts
-    double                                                  _inputPreprocessTime   = 0.0;
-    double                                                  _inputTransferTime     = 0.0;
-    double                                                  _executeTime           = 0.0;
-    double                                                  _outputTransferTime    = 0.0;
-    double                                                  _outputPostProcessTime = 0.0;
+    std::vector<std::shared_ptr<ngraph::runtime::Tensor>>   _inputTensors;
+    std::vector<std::shared_ptr<ngraph::runtime::Tensor>>   _outputTensors;
+    std::shared_ptr<ngraph::runtime::Executable>            _executable;
```

Comment:
should not it be created per executable network?

Reply:
I am sure that it is not thread safe to use it in multi threaded environment so it is better to have a copy in each infer request.